File: leetcode_cpp/roman_to_integer.cpp
class Solution {
public:
    unordered_map<char, int> roman_to_number = {
        {'I', 1},
        {'V', 5},
        {'X', 10},
        {'L', 50},
        {'C', 100},
        {'D', 500},
        {'M', 1000}
    };

    int romanToInt(string s) {
        int prior = roman_to_number.at(s[0]), total = 0, current;

        for(int i = 1; i < s.length(); ++i) {
            current = roman_to_number.at(s[i]);
            if (prior < current) {
                prior = current - prior;
            }
            else {
                total += prior;
                prior = current;
            }
        }
        total += prior;
        return total;
    }
};

File: leetcode_python/roman_to_integer.py
class Solution:
    # @return an integer

    roman_to_num = dict([('I', 1), ('V', 5), ('X', 10), ('L', 50), ('C', 100), ('D', 500), ('M', 1000)])

    def romanToInt(self, s):
        total = 0
        prior = self.roman_to_num[s[0]]
        for letter in s[1:]:
            current = self.roman_to_num[letter]
            if prior < current:
                prior = current - prior
            else:
                total += prior
                prior = current
        total += prior
        return total


        
